### PR TITLE
Add field storage schema definition for og_membership_reference field

### PR DESCRIPTION
**Pull request by damiankloip**
This should (I think) help with https://github.com/amitaibu/og/pull/11 however, I think we want this regardless.

### CODE REVIEW

**Comment by RoySegall** [comment]
Looks elegant and simple :) Will check and let you know.

**Comment by damiankloip** [comment]
Yes, let's see. Maybe too easy... ;)

**Comment by RoySegall** [comment]
Don't you know that there is a strait correlation between the number of hours spent in the bug to the size of the fix: As the hours spent in the bug increased the fix gets smaller.

**Comment by RoySegall** [comment]
Yap.. to easy :). Now I got this error:

> Drupal\Core\Config\Schema\SchemaIncompleteException: Schema errors for field.field.node.bcl9jeqa.og_group_ref with the following errors: field.field.node.bcl9jeqa.og_group_ref:settings.handler missing schema, field.field.node.bcl9jeqa.og_group_ref:settings.handler_settings missing schema

**Comment by damiankloip** [comment]
Ah, ok, try this now, just added some more.

**Comment by RoySegall** [comment]
Works like a charm. Closing.

**Comment by damiankloip** [comment]
@RoySegall See comment on https://github.com/amitaibu/og/pull/11. I think we should merge this PR and keep the history.

File: config/schema/og.schema.yml [hunk 1]
```diff
@@ -1,3 +1,22 @@
...
+
+field.field_settings.og_membership_reference:
+  type: mapping
+  label: 'Organic Groupds reference field settings'
+  mapping:
+    handler:
+      type: string
+      label: 'Reference method'
+    handler_settings:
+      type: entity_reference_selection.[%parent.handler]
```
Comment (amitaibu) [inline config/schema/og.schema.yml:17]: I didn't find in ER as-well- Where is the `parent.handler` value set ?

Plus admittedly I don't really understand what this part is doing (but I see it in ER so, it seems right)

**Comment by amitaibu** [comment]
:+1:

File: config/schema/og.schema.yml (hunk 1 above)
Comment (damiankloip) [inline config/schema/og.schema.yml:17]: This is for a dynamic type in the schema, see https://www.drupal.org/node/1905070 there is a cheat sheet PDF you can get on there which is actually pretty useful. This basically allows this configuration to stay the same, but change if the selection plugin were to change. We could just not have this for now, but seems like we might as well just leave it in. Makes switching it out easier/possible.

Comment (amitaibu) [inline config/schema/og.schema.yml:17]: :+1:

**Pull request merged**